ctru-rs/src/services/srv.rs

```diff
@@ -0,0 +1,27 @@
+//! Service API
+//!
+//! Not all APIs are wrapped in this module, since a lot are fundamentally unsafe.
+//! Most APIs should be used directly from `ctru-sys`.
+
+use crate::error::ResultCode;
+use ctru_sys::Handle;
+use std::time::Duration;
+
+/// Extension trait for [Handle]
+pub trait HandleExt {
+    /// Wait for an event to fire. If the timeout is reached, an error is returned. You can use
+    /// [`Error::is_timeout`] to check if the error is due to a timeout.
+    fn wait_for_event(self, timeout: Duration) -> crate::Result<()>;
+}
+
+impl HandleExt for Handle {
+    fn wait_for_event(self, timeout: Duration) -> crate::Result<()> {
+        unsafe {
+            ResultCode(ctru_sys::svcWaitSynchronization(
+                self,
+                timeout.as_nanos() as i64,
```

Comment:
This might be good to have a `try_into()?` for, since I think it's technically possible to pass something here that would wrap (not sure how that would behave in reality): https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=17885a20236d20e78d9a63df6aad11c7

Reply:
Updated